core/src/main/java/de/bwaldvogel/mongo/oplog/AbstractOplog.java

```diff
@@ -0,0 +1,30 @@
+package de.bwaldvogel.mongo.oplog;
+
+import java.time.Clock;
+
+import de.bwaldvogel.mongo.bson.Document;
+import io.netty.channel.Channel;
+
+public abstract class AbstractOplog implements Oplog {
+
+    protected final Clock clock;
+
+    public AbstractOplog(Clock clock) {
+        this.clock = clock;
+    }
+
+    @Override
+    public final void handleCommand(Channel channel, String databaseName, String command, Document query) {
```

Comment:
What is the value of this `AbstractOplog` ? Can `handleCommand` be a default method on the `Oplog` interface ?

Reply:
As it is right now you we do not need the abstract class, but later we may need it to implement common logic for the different oplog's.
I would say let's leave it for now and if turn out to be useless I'll remove it.

Reply:
Usually it is better to use strategy interfaces than one big base abstract class class.

Reply:
Since `AbstractOplog` is currently not needed I’ve removed it in 8e0540e6b5bf0a47b49cbb79936f4f0ac405d65b